cmd/kubectl/kubectl.go

```diff
@@ -16,18 +16,94 @@ limitations under the License.
 package kubectl
 
 import (
+	"fmt"
 	"os"
+	"os/exec"
+	"path"
+	"runtime"
+	"strings"
+	"syscall"
 
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 	"k8s.io/cli-runtime/pkg/genericclioptions"
 	kubectl "k8s.io/kubectl/pkg/cmd"
 
+	"github.com/k0sproject/k0s/internal/pkg/stringslice"
 	"github.com/k0sproject/k0s/pkg/config"
 )
 
 type CmdOpts config.CLIOptions
 
+// creates a temporary directory with a kubectl-symlink and prepends it to path. This is done to
+// allow kubectl plugins to call back to kubectl inside k0s
+func createTemp() (string, error) {
+	tempdir, err := os.MkdirTemp("", "k0s-kubectl")
```

Comment:
It is not uncommon that `/tmp` is mounted noexec, in which case things will fail.

I wonder if we can make a permanent symlink under datadir/bin

Reply:
Yeah maybe that would be better

Reply:
The kubectl subcommand will then only works when there's a k0s running on the host though.

Reply:
This will also require root for running kubectl in case datadir is in /var/lib.

Reply:
The tempdir has also the problem that defer won't fire so the tmp will not be cleaned 🤔

Reply:
Maybe just warn if there's no `kubectl` in PATH when launching a plugin and hint that you can make a symlink.

Reply:
Removed all the symlink trickery and replaced with simply a WARN when there's no kubectl in path.